src/Process/ParallelProcess.php

```diff
@@ -42,9 +42,9 @@ public function start(): self
         return $this;
     }
 
-    public function stop(): self
+    public function stop($timeout = 0): self
     {
-        $this->process->stop(10, SIGKILL);
+        $this->process->stop($timeout, SIGKILL);
```

Comment:
Why this change? When the process is marked as timed out by the pool, it must be stopped immediately and not wait another x-amount of time before being stopped.

Reply:
yes, I agree with you but pls check method in symfony component https://github.com/symfony/process/blob/7075b84559f3624c9f1fcde37c5af7a3e025ec8a/Process.php#L899
by default timeout is 10 sec.
but sorry may be I dont understood you, may be do you mean do like this `$this->process->stop(0, SIGKILL);` without possibility to manage?

Reply:
I'd keep it the way it is at 10. There's a reason that's the default in Symfony, from my testing I noticed that signals weren't properly processed if you'd kill a process with 0 seconds timeout.

Reply:
@brendt hey, thanks for your answer, but are you checked time for running process with stop child process with default timeout = 10 sec?

----------------------------------------
timeout for stop process 0 sec
root@70d0782122df:/var/www# time art cron:run:create_statistic_snapshots
real    4m42.866s
user    3m25.256s
sys     0m27.317s

----------------------------------------
timeout for stop process 10 sec
root@70d0782122df:/var/www# time art cron:run:create_statistic_snapshots
real    19m24.492s
user    5m15.208s
sys     1m19.971s


pls check it code https://prnt.sc/qnfuq3 if your process is running your will be wait while microtime(true) < $timeoutMicro

when will be call Pool:wait() each time when will be call $this->markAsTimedOut($process); script will be wait

this increases script execution time

Reply:
Let's set it to 0 then, if all tests still run, we're fine!